graphdatascience/query_runner/gds_arrow_client.py

```diff
@@ -89,19 +89,35 @@ def __init__(
         self._host = host
         self._port = port
         self._auth = auth
+        self._encrypted = encrypted
+        self._disable_server_verification = disable_server_verification
+        self._tls_root_certs = tls_root_certs
+        self._user_agent = user_agent
 
-        location = flight.Location.for_grpc_tls(host, port) if encrypted else flight.Location.for_grpc_tcp(host, port)
-
-        client_options: dict[str, Any] = {"disable_server_verification": disable_server_verification}
         if auth:
             self._auth_middleware = AuthMiddleware(auth)
-            if not user_agent:
-                user_agent = f"neo4j-graphdatascience-v{__version__} pyarrow-v{arrow_version}"
-            client_options["middleware"] = [AuthFactory(self._auth_middleware), UserAgentFactory(useragent=user_agent)]
-        if tls_root_certs:
-            client_options["tls_root_certs"] = tls_root_certs
 
-        self._flight_client = flight.FlightClient(location, **client_options)
+        self._flight_client = self._instantiate_flight_client()
+
+    def _instantiate_flight_client(self) -> flight.FlightClient:
+        location = (
+            flight.Location.for_grpc_tls(self._host, self._port)
+            if self._encrypted
+            else flight.Location.for_grpc_tcp(self._host, self._port)
+        )
+        client_options: dict[str, Any] = {"disable_server_verification": self._disable_server_verification}
+        if self._auth:
+            user_agent = f"neo4j-graphdatascience-v{__version__} pyarrow-v{arrow_version}"
+            if self._user_agent:
+                user_agent = self._user_agent
```

Comment:
so it could be
```suggestion
            user_agent = (
                self._user_agent
                if self._user_agent
                else f"neo4j-graphdatascience-v{__version__} pyarrow-v{arrow_version}"
            )
```

but it's not necessary. also maybe this will trip up the typer -- not sure.

Reply:
yeah looks much better this way